staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go

```diff
@@ -380,12 +399,23 @@ func TestWaitUntilFreshAndList(t *testing.T) {
 			return []string{pod.Spec.NodeName}, nil
 		},
 	})
-
+	fc := store.clock.(*testingclock.FakeClock)
 	// In background, update the store.
 	go func() {
+		for !fc.HasWaiters() {
+			time.Sleep(time.Millisecond)
```

Comment:
I doubt waiting 1ms really helps in overloaded envs.
If you want to wait for waiters - let's really try to ensure it.

Reply:
Copied from https://github.com/kubernetes/kubernetes/blob/dccc75705d8584122279db5a6923f0e26db85565/staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go#L489-L515

Reply:
Sorry - it's in the loop. I somehow missed that initially.